java/src/jmri/jmrix/loconet/LocoNetThrottle.java

```diff
@@ -380,7 +383,13 @@ public void notifyChangedSlot(LocoNetSlot pSlot) {
                     !((slotStatus & LnConstants.LOCOSTAT_MASK) == LnConstants.LOCO_IN_USE));
             slotStatus = newStat;
         }
-
+        
+        // It is possible that the slot status change we are being notified of
+        // is the slot being set to status COMMON. In which case the slot just
+        // got set to null. No point in continuing. In fact to do so causes a NPE.
+        if (slot == null)
```

Comment:
I've added the braces but left it as a 'short circuit' rather than put the rest of the code into a conditional. Functionally the two approaches are equivalent of course, and the encapsulation option is probably 'cleaner' but the short circuit makes it clearer to future maintainers what is happening, and why. 

I regard this check as a bit of a hack - it would be better to delay the loconet message setting the slot status to COMMON until we had no throttle objects listening. I might be able to get that working after a lot more time with the code, but having a solution that works now is  better than a potential more elegant solution at some unspecified time in the future (and maybe never).